### PR TITLE
gui: Add Windows taskbar progress

### LINKED ISSUE #14134: windows: Show sync progress in task bar
See initial discussion in #14090. This should probably be implemented using `QWinTaskbarProgress`, and could include some build system changes to include `winextras`.

### PR DESCRIPTION
**Pull request by luke-jr**
Rebase of #14137
Closes: #14134

### CODE REVIEW

**Comment by laanwj** [comment]
Concept ACK

File: src/qt/bitcoingui.cpp [hunk 1]
```diff
@@ -964,6 +969,11 @@ void BitcoinGUI::setNumBlocks(int count, const QDateTime& blockDate, double nVer
 
     tooltip = tr("Processed %n block(s) of transaction history.", "", count);
 
+#ifdef Q_OS_WIN
+    m_taskbar_button->setWindow(windowHandle());
```
Comment (laanwj) [inline src/qt/bitcoingui.cpp]: I don't think this is the place to do `setWindow`, every time `setNumBlocks` is called, move it to the initialization maybe?

Comment (luke-jr) [reply src/qt/bitcoingui.cpp]: Dunno, this apparently works as-is at least.

(I don't run Windows, so any major changes might be better off with someone else taking over the PR.)

**Comment by laanwj** [comment]
it fails the Travis compile:
```
compiling qwinjumplist.cpp

compiling qwinthumbnailtoolbar.cpp

compiling .moc/release/moc_qwintaskbarprogress.cpp

compiling .moc/release/moc_qwinthumbnailtoolbutton.cpp

compiling qwintaskbarbutton.cpp

qwinjumplist.cpp: In function ‘QString createArguments(const QStringList&)’:

qwinjumplist.cpp:99:66: error: invalid use of incomplete type ‘class QRegularExpression’

         tmp.replace(QRegularExpression(QLatin1String("(\\\\*)\"")), QLatin1String("\\1\\1\\\""));

                                                                  ^

In file included from /home/travis/build/bitcoin/bitcoin/depends/work/build/x86_64-w64-mingw32/qt/5.9.8-9cd2cbc4308/qtbase/include/QtCore/qstring.h:1:0,

                 from /home/travis/build/bitcoin/bitcoin/depends/work/build/x86_64-w64-mingw32/qt/5.9.8-9cd2cbc4308/qtbase/include/QtCore/../../src/corelib/kernel/qobject.h:47,

                 from /home/travis/build/bitcoin/bitcoin/depends/work/build/x86_64-w64-mingw32/qt/5.9.8-9cd2cbc4308/qtbase/include/QtCore/qobject.h:1,

                 from /home/travis/build/bitcoin/bitcoin/depends/work/build/x86_64-w64-mingw32/qt/5.9.8-9cd2cbc4308/qtbase/include/QtCore/../../src/corelib/animation/qabstractanimation.h:43,

                 from /home/travis/build/bitcoin/bitcoin/depends/work/build/x86_64-w64-mingw32/qt/5.9.8-9cd2cbc4308/qtbase/include/QtCore/qabstractanimation.h:1,

                 from /home/travis/build/bitcoin/bitcoin/depends/work/build/x86_64-w64-mingw32/qt/5.9.8-9cd2cbc4308/qtbase/include/QtCore/QtCore:5,

                 from ../../include/QtWinExtras/QtWinExtrasDepends:3:

/home/travis/build/bitcoin/bitcoin/depends/work/build/x86_64-w64-mingw32/qt/5.9.8-9cd2cbc4308/qtbase/include/QtCore/../../src/corelib/tools/qstring.h:80:7: note: forward declaration of ‘class QRegularExpression’

 class QRegularExpression;

       ^~~~~~~~~~~~~~~~~~
```
Looks like you're going to have to enable the regular expressions module for this…

**Comment by luke-jr** [comment]
Or disable qwinjumplist, which this doesn't even use.

**Comment by luke-jr** [comment]
I don't understand why gitian succeeds, but not Travis...

**Comment by MarcoFalke** [comment]
Gitian fails for the same reason

**Comment by luke-jr** [comment]
Ah, saw win32 builds above, but missed that it was just master...

**Comment by luke-jr** [comment]
For now, sed'd out the qwinjumplist stuff since it's completely unused.

**Comment by luke-jr** [comment]
@DrahtBot What's the point of the side-by-side when they're not the same on each side?

And why did the last build not even attempt?

**Comment by MarcoFalke** [comment]
@luke-jr How about that table? It might be more readable

**Comment by luke-jr** [comment]
Looks much nicer. Might make the .diffs two-column (if markdown supports it)

**Comment by jonasschnelli** [comment]
Concept ACK
Too bad this requires new dependencies and a bunch of ifdefs

File: configure.ac [hunk 2]
```diff
@@ -496,6 +496,7 @@ case $host in
      AC_CHECK_LIB([shlwapi],      [main],, AC_MSG_ERROR(libshlwapi missing))
      AC_CHECK_LIB([iphlpapi],      [main],, AC_MSG_ERROR(libiphlpapi missing))
      AC_CHECK_LIB([crypt32],      [main],, AC_MSG_ERROR(libcrypt32 missing))
+     AC_CHECK_LIB([dwmapi],      [main],, AC_MSG_ERROR(libdwmapi missing))
```
Comment (laanwj) [inline configure.ac]: Do you know what is the first release of windows to have `dwmapi`? Is this a recent addition? (if it's windows 7 or older it's fine)

Comment (luke-jr) [reply configure.ac]: https://docs.microsoft.com/en-us/windows/win32/api/dwmapi/nf-dwmapi-dwmsetwindowattribute suggests at least Vista has it.

**Comment by fanquake** [comment]
Concept ~0. I've closed my original issue (#14134), and don't really think that this change and the additional dependencies is really worthwhile.

**Comment by luke-jr** [comment]
Closing due to lack of interest

**Pull request closed without merging**